### ISSUE TITLE
Figure out a better way to verify the e2e test

### ISSUE THREAD

**Issue by capri-xiyue** [abcxyz/lumberjack#346]
### TL;DR

Previously, we rely on single filed to verify the e2e test, see details here: https://github.com/abcxyz/lumberjack/pull/340#issuecomment-1334172512

To make the e2e test more reliable, we merged https://github.com/abcxyz/lumberjack/pull/345#pullrequestreview-1210804849 to rely on multiple fields to verify the e2e test.

@sethvargo added a comment in https://github.com/abcxyz/lumberjack/pull/345#discussion_r1043842668, we can try to figure out a better way to verify the e2e test. I think to serialize the result into a structure and then assert the whole structure as a diff could be too much. A lot of fields related to `justification` in the BQ data are quite complex. I think we can serialize the result into a structure and assert some essential fields like `jsonPayload.service_name` and `jsonPayload.authentication_info.principal_email`

**Comment by yolocs** [comment]
Background:

- https://github.com/abcxyz/lumberjack/blob/main/integration/testrunner/main_test.go is the entry point for our integration test
- https://github.com/abcxyz/lumberjack/blob/main/.github/workflows/ci.yml is the CI workflow, in which we run the integration tests
- General test flow:
  1. Deploy test apps
  2. In test, send requests to the test apps
  3. The test apps to write audit logs that will be routed to BQ
  4. In test, we check if expected logs appear in the BQ
- The problem we're having: we only pick certain fields from BQ to check. But it's better if we check the whole log and intentionally ignore certain fields.

Steps:

- Learn what data will end up in BQ.
- Learn how to extract a BQ row into a `LogEntry`. Here is how we query BQ today: Here is how we query BQ https://github.com/abcxyz/lumberjack/blob/main/integration/testrunner/db.go#L60
- In test, instead of checking if a log entry exists in BQ (https://github.com/abcxyz/lumberjack/blob/main/integration/testrunner/db.go#L87), we should extract the log and compare with an expected log entry.
